test/unit/cmap/message_stream.test.js

```diff
@@ -19,6 +19,34 @@ function bufferToStream(buffer) {
 }
 
 describe('Message Stream', function () {
+  context('when the stream uses the streaming protocol', function () {
+    const data = Buffer.from(
+      '370000000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000' +
+        '370000000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000' +
+        '370000000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000' +
+        '370000000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000',
+      'hex'
+    );
+
+    it('only reads the last message in the buffer', function (done) {
+      const inputStream = bufferToStream(data);
+      const messageStream = new MessageStream();
+      messageStream.isStreamingProtocol = true;
+
+      messageStream.once('message', msg => {
+        msg.parse();
+        expect(msg)
+          .to.have.property('documents')
+          .that.deep.equals([{ [LEGACY_HELLO_COMMAND]: 1 }]);
```

Comment:
might it be better to use a buffer that uses a regular hello command instead of legacy since the legacy one is what we're aiming to eventually remove?

Reply:
I've refactored all the tests now to make it more clear what's happening, including a utility function to generate OP_MSG buffers from a document. The document I use now is a simple response from a normal hello to test.